src/main/java/app/controller/graphicsEngine/Ray.java

```diff
@@ -1,13 +1,15 @@
 package app.controller.graphicsEngine;
 
 import app.controller.linAlg.Vector;
+import app.model.agents.Team;
 import app.view.simulation.Info;
 import javafx.scene.canvas.GraphicsContext;
 import javafx.scene.paint.Color;
 import lombok.Getter;
 
 public class Ray
 {
+    @Getter private Team agentTeam;
```

Comment:
I think you should make a default null value for this variable, as the `Ray(Vector u, Vector v)` constructor doesn't initialise it, this may cause a Null Pointer in the future that will be annoying to debug.

Reply:
Other than that, nice implementation!